README.md

```diff
@@ -586,22 +586,46 @@ The encrypted binary will have the following structure:
 - Padding to ensure the encrypted length is a multiple of 4 words
 - Signature metadata block
 
-The AES key must be provided as a .bin file of the 256 bit AES key to be used for encryption.
+The AES key can either be provided as a 1024-bit (128 byte) 4-way key share or as a 256-bit (32 byte) AES key. In the latter case, a key share will be generated for you.
+
+The 4-way key share should be generated by creating a random .bin file of length 128 bytes and the AES key will be derived from that. With the words stored in the file as A[0], B[0], C[0], D[0], A[1], B[1], etc., C[7], D[7], word i of the key X is X[i] = A[i] ^ B[i] ^ C[i] ^ D[i].
+
+Alternatively an AES key can be provided directly, as either a 32 byte .bin file or as a string of 64 hexadecimal characters (eg 0x0123456789abcdeffedcba98765432100123456789abcdeffedcba9876543210). Picotool will generate a random key share for that key, by generating 3 random shares (A, B & C) and calculating a 4th share (D) to match the provided key (X), with each word i of the 4th share calculated as D[i] = X[i] ^ A[i] ^ B[i] ^ C[i].
+
+The encryption/decryption code also salts the IV (initialisation vector) so it's not stored in plaintext in the binary. This requires a per-device IV salt in OTP, which should be a 128-bit (16 byte) random value - this can be provided as either a 16 byte .bin file, or a string of 32 hexadecimal characters, similar to the AES key. This per-device salt will be XORed with the IV stored in the binary, to give the IV used by the decryption code.
 
 ```text
 $ picotool help encrypt
 ENCRYPT:
     Encrypt the program.
 
 SYNOPSIS:
-    picotool encrypt [--quiet] [--verbose] [--hash] [--sign] <infile> [-t <type>] [-o <offset>] <outfile> [-t <type>] <aes_key> [-t <type>]
-                [<signing_key>] [-t <type>]
+    picotool encrypt [--quiet] [--verbose] [--embed] [--fast-rosc] [--use-mbedtls] [--otp-key-page <page>] [--hash] [--sign] <infile>
+                [-t <type>] [-o <offset>] <outfile> [-t <type>] <aes_key> <iv_otp> <signing_key> <otp>
 
 OPTIONS:
         --quiet
             Don't print any output
         --verbose
             Print verbose output
+        --embed
+            Embed bootloader in output file
+        --fast-rosc
+            Use ~180MHz ROSC configuration for embedded bootloader
+        --use-mbedtls
+            Use MbedTLS implementation of embedded bootloader
+        --otp-key-page
+            Specify the OTP page storing the AES key (IV salt is stored on the next page)
+        <page>
+            OTP page (default 30)
+        <aes_key>
+            AES Key Share or AES Key
+        <iv_otp>
+            IV OTP Salt
+        <signing_key>
+            Signing Key file
+        <otp>
+            File to save OTP to (will edit existing file if it exists)
```

Comment:
I see that in `main.cpp` this has been updated to "JSON file to save OTP to (will edit existing file if it exists)", so I guess you might want to use #220 here? :grinning:

Reply:
Yep, I think I'll merge #220 into develop, then merge develop into here again and run it